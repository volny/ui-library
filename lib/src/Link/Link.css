.link {
  composes: underline from "../typography.css";
  composes: darkGrey70 from "../colors.css";
  cursor: pointer;
  outline: none;
}

.link:hover,
.link:focus {
  /* TODO: can't compose on pseudoselectors */

  /* https://github.com/css-modules/css-modules/issues/80 */

  /*   composes: darkGrey from "../colors.css" */

  color: #373737;
}
