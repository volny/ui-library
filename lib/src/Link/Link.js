import * as React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import styles from './Link.css'
import layout from '../layout.css'

const Link = ({ children, inline, target, href, onClick }) => (
  <a
    className={cx(styles.link, inline ? '' : layout.block)}
    href={href}
    onClick={onClick || null}
    target={target ? `_${target}` : null}
  >
    {children}
  </a>
)

Link.defaultProps = {
  inline: false,
  target: null,
}

Link.propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string,
  inline: PropTypes.bool,
  onClick: PropTypes.func,
  target: PropTypes.oneOf([null, 'self', 'blank']),
}

export default Link
