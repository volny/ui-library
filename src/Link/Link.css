.link {
  composes: underline from "../typography.css";
  composes: darkGrey70 from "../colors.css";
  outline: none;
}

.link:hover,
.link:focus {
  /* composes: darkGrey from "../colors.css" */
}

/* .block { */
/*   composes: block from "../Layout.css"; */
/* } */
