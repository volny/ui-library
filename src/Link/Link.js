import * as React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import colors from '../colors.js'

class Link extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    href: PropTypes.string.isRequired,
    inline: PropTypes.bool,
    onClick: PropTypes.func,
    target: PropTypes.oneOf([null, 'self', 'blank']),
  }

  handleClick(event) {
    const { href, onClick } = this.props
    if (onClick && href) {
      onClick({ event })
    }
  }

  render() {
    const { children, inline = false, target = null, href } = this.props
    const linkTarget = target ? `_${target}` : null

    const Styled = styled.a`
      text-decoration: underline;
      color: ${colors.darkGrey70};
      outline: none;
      &:hover,
      &:focus {
        color: ${colors.darkGrey};
      }
      ${inline || 'display: block;'};
    `

    return (
      <Styled href={href} onClick={this.handleClick} target={linkTarget}>
        {children}
      </Styled>
    )
  }
}

export default Link
