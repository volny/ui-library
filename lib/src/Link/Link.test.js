/**
 * @jest-environment node
 */

/* eslint-env jest */
import React from 'react'
import { create } from 'react-test-renderer'

import Link from './Link'

test('Link creates a link tag', () => {
  const tree = create(<Link href="https://example.com">Link</Link>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Link target blank works', () => {
  const tree = create(
    <Link href="https://example.com" target="blank">
      Link
    </Link>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Link onClick prop works', () => {
  const tree = create(<Link onClick={() => 'hola'}>Link</Link>).toJSON()
  expect(tree).toMatchSnapshot()
})
