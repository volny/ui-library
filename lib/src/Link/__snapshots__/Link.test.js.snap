// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link creates a link tag 1`] = `
<a
  className="link block"
  href="https://example.com"
  onClick={null}
  target={null}
>
  Link
</a>
`;

exports[`Link onClick prop works 1`] = `
<a
  className="link block"
  href={undefined}
  onClick={[Function]}
  target={null}
>
  Link
</a>
`;

exports[`Link target blank works 1`] = `
<a
  className="link block"
  href="https://example.com"
  onClick={null}
  target="_blank"
>
  Link
</a>
`;
